crates/agglayer-node/src/kernel/mod.rs

```diff
@@ -304,7 +315,35 @@ where
 
         let signer = signed_tx
             .signer()
-            .map_err(|e| SignatureVerificationError::CouldNotRecoverSigner(e))?;
+            .map_err(SignatureVerificationError::CouldNotRecoverTxSigner)?;
+
+        if signer != sequencer_address {
+            return Err(SignatureVerificationError::InvalidSigner {
+                signer,
+                trusted_sequencer: sequencer_address,
+            });
+        }
+
+        Ok(())
+    }
+
+    /// Verify that the signer of the given [`Certificate`] is the trusted
+    /// sequencer for the rollup id it specified.
+    #[instrument(skip(self), level = "debug")]
+    pub(crate) async fn verify_certificate_signer(
```

Comment:
I'm not sure it's a good idea to put this into the `kernel` component. It is (if I understand it correctly) basically a backwards compatibility layer for the previous version of agglayer. Now placing it here allows you to reuse some code but it would better be factored out imho. Not sure how pressing this is though. Or am I mistaken here @Freyskeyd?